cmd/ak/cmd/runtimes/run.go

```diff
@@ -0,0 +1,168 @@
+package runtimes
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/hexops/gotextdiff"
+	"github.com/hexops/gotextdiff/myers"
+	"github.com/hexops/gotextdiff/span"
+	"github.com/spf13/cobra"
+	"golang.org/x/tools/txtar"
+
+	"go.autokitteh.dev/autokitteh/cmd/ak/common"
+	"go.autokitteh.dev/autokitteh/internal/kittehs"
+	"go.autokitteh.dev/autokitteh/sdk/sdkruntimes/sdkbuildfile"
+	"go.autokitteh.dev/autokitteh/sdk/sdkservices"
+	"go.autokitteh.dev/autokitteh/sdk/sdktypes"
+)
+
+var (
+	path       string
+	tmo        time.Duration
+	txtarFile  bool
+	emitValues bool
+	test       bool
+)
+
+var runCmd = common.StandardCommand(&cobra.Command{
+	Use:     "run <build file|program file> [--txtar]",
+	Short:   `Run a program`,
+	Aliases: []string{"r"},
+	Args:    cobra.ExactArgs(1),
+
+	RunE: func(cmd *cobra.Command, args []string) error {
+		f, err := os.OpenFile(args[0], os.O_RDONLY, 0)
+		if err != nil {
+			return fmt.Errorf("open: %w", err)
+		}
+
+		defer f.Close()
+
+		var b *sdkbuildfile.BuildFile
+
+		var expectedPrints string
+
+		if txtarFile || test {
```

Comment:
The logical flow of this code is overly complex. You should make it simpler to trace.

Reply:
agree. will do in the future. this is not easy :/